### ISSUE TITLE
GRC RFNoC Digital Upconverter Block Python Code Generation

### ISSUE THREAD

**Issue by jmingraham** [EttusResearch/gr-ettus#49]
GRC does not generate valid Python code for the RFNoC Digital Upconverter Block.
GRC adds an additional argument (5 instead of 4) when instantiating the ettus.rfnoc_duc block.
The method set_freq is run before set_input_rate in the generated Python code. When this is done, Python complains that it is no longer possible to set the input_rate. The method set_freq also does not appear to change the frequency shift of the ettus.rfnoc_duc.

**Comment by sma7c** [comment]
We've replicated this issue with our X310 integration efforts. needs to be resolved or there's no way to use GRC with RFNOC radio tx.

**Comment by nlhnt** [comment]
We've encountered this issue as well today.

**Comment by blaszki** [comment]
Line 11 in [ettus_rfnoc_duc.block.yml](https://github.com/EttusResearch/gr-ettus/blob/1038c4ce5135a2803b53554fc4971fe3de747d9a/grc/ettus_rfnoc_duc.block.yml#L11) is a problem: 
```
make: |-
    ettus.rfnoc_duc(
        self.rfnoc_graph,
        ${num_chans},
        uhd.device_addr(${block_args}),
        ${device_select},
        ${instance_index})
```
Workaround: you can delete it from prefix/share/gnuradio/grc/blocks/ettus_rfnoc_duc.block.yml

**Comment by mbr0wn** [comment]
Resolved by e3a6e8008d945847ec1ec4d0f50267dc8d3f8dbe
